git/github.go

```diff
@@ -0,0 +1,174 @@
+package git
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/weaveworks/flux/http/error"
+	"io"
+	"net/http"
+)
+
+const (
+	deployKeyName = "flux-generated"
+)
+
+// ******************** Domain ********************
+
+type owner struct {
+	name  string
+	token string
+}
+
+type repo struct {
+	name string
+}
+
+type deployKey struct {
```

Comment:
Actually I'm not sure why a deploy key isn't just a value, with Github having the operations to get, create, etc.